src/modules/directory.rs

```diff
@@ -109,22 +117,46 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
             if ((num_segments_after_root - 1) as i64) < config.truncation_length {
                 let root = repo_path_vec[0];
                 let before = dir_string.replace(&contracted_path, "");
-                [prefix + &before, root.to_string(), after_repo_root]
+                path_vec[BEFORE_ROOT_PATH] = prefix + &before;
+                path_vec[REPO_ROOT] = root.to_string();
+                path_vec[PATH] = after_repo_root;
             } else {
-                ["".to_string(), "".to_string(), prefix + &dir_string]
+                path_vec[PATH] = prefix + &dir_string;
             }
         }
-        _ => ["".to_string(), "".to_string(), prefix + &dir_string],
+        _ => {
+            path_vec[PATH] = prefix + &dir_string;
+        }
     };
 
+    (path_vec[PATH_BASE], path_vec[LAST_COMPONENT]) =
+        path_vec[PATH]
+            .rfind('/')
+            .map_or(("".to_string(), path_vec[PATH][..].to_string()), |pos| {
+                (
+                    path_vec[PATH][..pos].to_string(),
+                    path_vec[PATH][pos + 1..].to_string(),
+                )
+            });
+
+    if path_vec[LAST_COMPONENT] != home_symbol
```

Comment:
Would be nice to have this factored out into a named function (or an `impl` on a struct which contains this information) because I can't work out what this is supposed to do.